did_peer/src/peer_did/peer_did.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{error::DidPeerError, numalgos::numalgo2::generate_numalgo2};
+use did_doc::schema::did_doc::DidDocument;
+use did_doc_sov::extra_fields::ExtraFieldsSov;
+use did_parser::Did;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use super::{numalgo::Numalgo, transform::Transform};
+
+// TODO: This regex does not cover peer:did:3
+static PEER_DID_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^did:peer:(([01](z)([1-9a-km-zA-HJ-NP-Z]{5,200}))|(2((.[AEVID](z)([1-9a-km-zA-HJ-NP-Z]{5,200}))+(.(S)[0-9a-zA-Z=]*)?)))$").unwrap()
+});
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct PeerDid {
+    did: Did,
+    numalgo: Numalgo,
+    transform: Option<Transform>,
+}
+
+impl PeerDid {
+    pub fn parse<T>(did: T) -> Result<Self, DidPeerError>
```

Comment:
But I've realized that if you'd want to call resolve method right here, the resolve method right now is parametrized with options, which would make things slightly trickier, especially if we would want to store the resolved DidDoc into `PeerDid` itself. Hmm.

Well I suppose the status quo is better after all.

Reply:
Isn't the regex sufficient to validate the DID? Moreover, you might want to use the DID without resolving it, so resolving it when parsing seems redundant. Also, resolution is numalgo-specific and may not be possible (e.g. numalgo 3).

Reply:
well, if some of the key fragments in peer-did did is nonsense, altough it is still a did, it is not a peer did. The only way to check that peer-did is valid peer-did is to actually resolve it (which is technically just decoding) (at least for numalgo 2)

Anyway, you can resolve this thread (no pun intended :-D)

Reply:
I guess one might distinguish valid unresolvable peer dids from invalid peer dids... But I see what you mean. I would leave the user to try resolving the DID in addition to parsing it instead of resolving it for him when parsing (which seems a little unexpected). Also, there is the issue with resolution options... Resolving - the thread for now :).